script.js

```diff
@@ -0,0 +1,987 @@
+        this.createButtonRow(keyboard, buttonArrayRow4);
+        this.createButtonRow(keyboard, buttonArrayRow5);
+
+        const description = document.createElement('p');
+        description.classList.add('description');
+        description.innerText = 'Клавиатура создана в операционной системе Windows';
+        wrapper.append(description);
+
+        const changeLang = document.createElement('p');
+        changeLang.classList.add('language');
+        changeLang.innerText = 'Для переключения языка комбинация: левыe Alt + Shift';
+        wrapper.append(changeLang);
+
+
+        document.addEventListener('mousedown', this.mouseDownEvent.bind(this)); // Должен появляться класс active
+        document.addEventListener('mouseup', this.mouseUpEvent.bind(this));
+
+        document.addEventListener('keydown', this.keyDownEvent.bind(this));
+        document.addEventListener('keyup', this.keyUpEvent.bind(this));
+
+        this.changeLanguage();
+
+    }
+
+    clickButton (event) {
+
+        let keyCode;
+
+        if (!event.target.dataset.button) {
+            let parentDiv = event.target.closest('div');
+            keyCode = parentDiv.dataset.button;
+        } else {
+            keyCode = event.target.dataset.button;
+        }
+
+        switch (keyCode) {
+            case 'Backspace':
+                this.backSpaceHandler()
+                break;
+            case 'Tab':
+                this.tabHandler()
+                break;
+            case 'Delete':
+                this.delHandler()
+                break;
+            case 'CapsLock':
+                this.capsHandler(event)
+                break;
+            case 'Enter':
+                this.enterHandler()
+                break;
+            case 'ShiftLeft':
+                this.shiftHandler(event)
+                break;
+            case 'ShiftRight':
+                this.shiftHandler(event)
+                break;
+            case 'ControlLeft':
+                break;
+            case 'MetaLeft':
+                break;
+            case 'AltLeft':
+                break;
+            case 'AltRight':
+                break;
+            case 'ControlRight':
+                break;
+            default:
+                this.printText(event)
+        }
+
+    }
+
+    printText(event) {
+        const key_child_span = document.querySelectorAll(`.${event.target.dataset.button} span span `);
+
+        if (key_child_span.length > 0) {
+            /* Target DIV и значет мы ищем чего написать */
+            key_child_span.forEach((item) => {
+                if (!item.classList.contains('hidden')) {
+                    this.setTextToInput(item.textContent)
+                }
+            });
+        } else {
+            /* Мы уже находимя на букве и можем ее писать */
+            this.setTextToInput(event.target.textContent)
+        }
+    }
+
+    tabHandler() {
+        this.setTextToInput("\t")
+    }
+
+    backSpaceHandler() {

```

Comment:
looks like this method will not work correctly for the selected area as it respects only selection start.
Please update it based on delHandler() implementation (deletion of selection probably could be moved to a separate method).

Reply:
done